src/classes/OpportunityDataFactory.cls

```diff
@@ -0,0 +1,23 @@
+public with sharing class OpportunityDataFactory extends DataFactory {
+	
+	AccountDataFactory adf = new AccountDataFactory(Account.SObjectType);
+	
+	 public OpportunityDataFactory(SObjectType objectType) {
+        super(objectType);
+    }
+    
+    public SObject build() {
+    	return build((Account) adf.getDefaultInstance());
+    }
+    
+    public SObject build(Account a) {
+    	Opportunity o = new Opportunity();
```

Comment:
This variable doesn't seem to have a very meaningful name. I suggest to give this variable a concise but meaningful name that suggest its purpose.